### PR TITLE
ROU-4196: Implement Search component with new architecture

### PR DESCRIPTION
**Pull request by bmarcelino-fe**
This PR is for revamp of Search pattern with the new architecture and accessibility improvements

### What was done
- HTML structure reviewed
- CSS improvements on animations and pattern behaviors
- Created API to enable native behavior, get states and update AriaLabel for translations purposes
- Implemented the new architecture on pattern

### Checklist
-   [X] tested locally
-   [X] documented the code
-   [X] clean all warnings and errors of eslint
-   [ ] requires changes in OutSystems (if so, provide a module with changes)
-   [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

File: src/scripts/OutSystems/OSUI/Patterns/SearchAPI.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+				const search = this.GetSearchById(searchId);
+
+				search.enableNativeBehavior();
+			},
+		});
+
+		return result;
+	}
+
+	export function UpdateGlassButtonAriaLabel(searchId: string, ariaLabel: string): string {
```
Comment (OS-giulianasilva) [inline src/scripts/OutSystems/OSUI/Patterns/SearchAPI.ts:153]: Missing a comment explaining the function as we have for the others

File: src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+			// Create HTML element for native behavior
+			this._searchGlassButton = document.createElement(OSFramework.OSUI.GlobalEnum.HTMLElement.Div);
+			this._searchGlassButton.classList.add(Enum.CssClass.PatternGlassButton);
+			this.selfElement.appendChild(this._searchGlassButton);
+
+			// Set A11Y properties to elements
+			Helper.A11Y.TabIndexFalse(this._searchInput);
+			Helper.A11Y.TabIndexTrue(this._searchGlassButton);
+			Helper.A11Y.AriaExpandedFalse(this._searchGlassButton);
+			Helper.A11Y.AriaLabel(this._searchGlassButton, 'Open Search');
```
Comment (OS-giulianasilva) [inline src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts:45]: Can we add this string into the Enum file?

File: src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+				// Set A11Y properties
+				Helper.A11Y.TabIndexFalse(this._searchInput);
+				Helper.A11Y.AriaExpandedFalse(this._searchGlassButton);
+				Helper.A11Y.AriaLabel(this._searchGlassButton, 'Open Search');
+
+				this._isOpen = false;
+			} else {
+				// Focus on search input if is empty
+				if (this._searchInput && this._searchInput.value === '') {
```
Comment (OS-giulianasilva) [inline src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts:92]: Why move the focus only if the input is empty?

File: src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+					}, 300);
+				}
+
+				// Add the isOpen class and update the state variable
+				this.selfElement.classList.add(Enum.CssClass.PatternIsOpen);
+
+				// Set A11Y properties
+				Helper.A11Y.TabIndexTrue(this._searchInput);
+				Helper.A11Y.AriaExpandedTrue(this._searchGlassButton);
+				Helper.A11Y.AriaLabel(this._searchGlassButton, 'Close Search');
```
Comment (OS-giulianasilva) [inline src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts:104]: Can we add this string into the Enum file?

**Review by OS-giulianasilva — APPROVED** [review]

File: src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts (hunk 3 above)
Comment (bmarcelino-fe) [reply src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts:92]: This is a old use case of the search component and I kept the same on this new one. Basically they can search and keep the value on the input and navigate between data inside the screen.

File: src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts (hunk 4 above)
Comment (bmarcelino-fe) [reply src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts:104]: Yes we can. I will change it.

File: src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts (hunk 2 above)
Comment (bmarcelino-fe) [reply src/scripts/OSFramework/OSUI/Pattern/Search/Search.ts:45]: Yes we can. I will change it.

File: src/scripts/OutSystems/OSUI/Patterns/SearchAPI.ts (hunk 1 above)
Comment (bmarcelino-fe) [reply src/scripts/OutSystems/OSUI/Patterns/SearchAPI.ts:153]: Done

**Review by OS-giulianasilva — APPROVED** [review]

**Review by OS-giulianasilva — APPROVED** [review]

**Review by joselrio — CHANGES_REQUESTED** [review]

**Comment by bmarcelino-fe** [comment]
This PR will be closed in order to remove the support of native behavior. However, if in the future we decided to implement this behavior, here is the PR with all the code.

**Pull request closed without merging**